Bay-like cryotube pods

nicbn opened the pull request:
![hi world](https://user-images.githubusercontent.com/20824501/27188882-fe0c711a-51c5-11e7-84cf-34c57c5e175f.png)
![howd](https://user-images.githubusercontent.com/20824501/27188883-fe3c7356-51c5-11e7-921e-960390e55d77.png)
![alan](https://user-images.githubusercontent.com/20824501/27209693-a43eef2e-5223-11e7-8c7d-0436a42e96c3.png)

[Changelogs]:
[]:
:cl:
imageadd: Changed cryo sprites to Bay cryopods, which show the person inside of them.
/:cl:

[why]:
Cryotube pod sprites shows up the sprite of who's inside it, looking way more neat.

nicbn committed:
Add files via upload

nicbn committed:
Add files via upload

nicbn committed:
Add files via upload

GunHog commented:
Now THIS is an improvement!!

ChangelingRain commented:
It loses the glow at the top when on?

ChangelingRain commented:
Also I don't like the base, it's a weird color compared to the rest of our on-station sprites.

WJohn commented:
This is pretty cool, but I'm not a huge fan of the colors either. They're too dark, and should definitely be a very bright shining color when on.

PKPenguin321 commented:
Yeah the base and the colors are weird but the effect is dope and I'm willing to sacrifice the old sprite for it

nicbn commented:
The color is the same of the VG sprites. What I changed was some alpha that makes the cryo look like glass and I added a open state for it (no cover). I also added pipes in the bottom of the sprite and, of course, added some code to make it work.

DoctoreMad commented:
>VG

[Snicker]

WJohn commented:
The open state is not very obvious at all, to be honest.

RandomMarine commented:
While the fact that it looks overall _better_ is undeniable, I don't really like seeing the person inside of it. I feel obfuscation is a key point of SS13.

I.E. this kills tactical hiding inside of cryo tubes to elude the person looking for you.

PKPenguin321 commented:
Hmm. You have a point

GunHog commented:
What if the sprite is obscured only if it is actually on?

FantasticFwoosh commented:
Suggestion and/or nerf

> The color of the tube should change to the colors of the reagent being currently inserted into it.

GunHog commented:
^ +1 to that idea

Kromgar commented:
How am i supposed to fill cryo with plasmaman toxin now?

RandomMarine commented:
It would like nice but it would make sabotaging cryo even less viable than it currently is.

Kromgar commented:
We should remove cryo telling us what's in the mix you need science goggles to do so

nicbn commented:
> The color of the tube should change to the colors of the reagent being currently inserted into it.

If this gets pulled I'll start working on this. Please don't talk about this here.

nicbn commented:
> The open state is not very obvious at all, to be honest.

Not for now, but if we use it in the game, we will familiarize ourselves with it eventually.
The alpha of the open state is lower in the glass too, that way we can say if it's closed by seeing if it's "too opaque" (or by trying to take someone inside it and failing).

> While the fact that it looks overall better is undeniable, I don't really like seeing the person inside of it. I feel obfuscation is a key point of SS13.
> I.E. this kills tactical hiding inside of cryo tubes to elude the person looking for you.

There are DNA scanners, lockers and sleepers in the medbay that can be used. The key point of cryotubes is to help doctors, and seeing who's inside it helps the doctors a lot.

RandomMarine commented:
And a key point of many objects in SS13 in the game is that they can be misused and sabotaged. Don't kill that angle through QoL changes.

nicbn commented:
But hiding in the cryos already is a bad move. Think a little bit, you need to wait ~30 seconds to escape the cryo without someone opening it for you. If someone is chasing you and you hide in the cryotube, he can even turn it on and you'll fall asleep, then you are doomed.
Also if you want to sabotage, you could sabotage a cryo knowing who's inside, making the sabotage easier.

Chronitonity commented:
>Not for now, but if we use it in the game, we will familiarize ourselves with it eventually.

Itd just straight up better to have the open state be more obvious instead of saying "Well yeah we'll get used to it"

nicbn committed:
Lowered open state glass alpha

WJohn commented:
Alpha is not the problem. It needs to visually show that there is a door that has been opened. Our current ones imply that the front half comes off in some way, but is moved in place. Maybe do the same there, but show that it slides into the ground or splits down the middle and slides into the tube or something.

nicbn commented:
I updated the PR screenshots so you can see how it looks now. I also added the light detail on the on state.

darkpaladin109 commented:
I love this, but I do agree with the concerns raised about not being able to hide in tubes anymore. Maybe we could compromise by showing only showing a silhouette when it's not on? That way, you could still theoretically hide in a tube.

As for the Fwoosh's suggestion to make the tube change color based on the chems, maybe we could add a way for people to obfuscate the color?

Bawhoppen commented:
Liberty station had this and I made a PR to port it awhile back if we want to use our current sprites instead

![2hdnlof](https://user-images.githubusercontent.com/6510293/27021249-b7fd8472-4f0c-11e7-9d27-608542af404f.gif)

nicbn commented:
Well I like the bay sprites, they are rounder and cool.
I'll be updating it soon, and I'll add a video to the pull showing it work.

Cyberboss commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -87,17 +90,46 @@
+
+	if(occupant)
+		occupant_overlay = mutable_appearance(occupant.icon, occupant.icon_state)
+		occupant_overlay.overlays = occupant.overlays
```
occupant_overlay.copy_overlays(occupant)

Cyberboss commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -87,17 +90,46 @@
+		add_overlay(occupant_overlay)
+		if(on && is_operational() && !bobbing)
+			var/bobup = TRUE
+			spawn(0) // Time to bob
```
Make this a seperate proc like /obj/machinery/atmospherics/components/unary/cryo_cell/proc/cryo_animation

Cyberboss commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -87,17 +90,46 @@
+						occupant_overlay.pixel_y--
+					add_overlay(occupant_overlay)
+					add_overlay("cover-on")
+					sleep(7)
```
Use id'd timers instead of sleeps

Cyberboss commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -26,6 +26,9 @@
 	var/radio_key = /obj/item/device/encryptionkey/headset_med
 	var/radio_channel = "Medical"
 
+	var/mutable_appearance/occupant_overlay
```
You regenerate this overlay every call to update_icon. Is it needed up here?

nicbn replied:
All your suggestions will be merged in the next commit.

nicbn committed:
Now cryos have open icon

nicbn committed:
and the code is fixed

nicbn commented:
PR updated with new pics.

ChangelingRain commented:
This might look very weird if you put an alien queen or someone under the effects of growth serum in it, since they'd clip out of the pod.

Cyberboss commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -92,44 +93,44 @@
+		occupant_overlay.pixel_y--
+	add_overlay(occupant_overlay)
+	add_overlay("cover-on")
+	addtimer(CALLBACK(src, .proc/run_bob_anim, anim_up, occupant_overlay), 7)
```
Should be TIMER_UNIQUE

nicbn committed:
oops

nicbn committed:
wrong path

nicbn committed:
oops2

nicbn commented:
> This might look very weird if you put an alien queen or someone under the effects of growth serum in it, since they'd clip out of the pod.

Monkeys and alans will have specific sprites for cryogenics so the tails won't "come out" of the cryotube, and other mobs will have a generic sprite for it.

nicbn committed:
Custom sprites for some mobs inside cryo

nicbn committed:
More code

nicbn commented:
Also it picks the mob icon and initiates a mutable appearance for it, so people won't be shown with effects such as growth serum.

ChangelingRain commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -87,17 +91,72 @@
+			else if(istype(occupant, /mob/living/carbon/alien/humanoid/hunter)) // Hunter
+				occupant_overlay = mutable_appearance(CRYOMOBS, "alienh")
+
+			else if(istype(occupant, /mob/living/carbon/alien/humanoid/sentinel)) // Sentinel
```
please use the is helper macros for these

nicbn replied:
Sorry about the newbieness, but... What is it and how do I use it?

nicbn replied:
Oh found it.

nicbn replied:
Sure

nicbn replied:
Should I create new is helpers for alien types or just use istype?

ChangelingRain commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -87,17 +91,72 @@
+				occupant_overlay = mutable_appearance(CRYOMOBS, "alienq")
+
+			else if(istype(occupant, /mob/living/carbon/alien/humanoid/hunter)) // Hunter
+				occupant_overlay = mutable_appearance(CRYOMOBS, "alienh")
```
I'd like to note that I _really_ don't like this as a solution and would kind of prefer that you crop the mutable appearance to fit the tube, even though that might look weird sometimes.

Maybe also shift large icons like alien queens/praetorians down a little.

nicbn replied:
I used this way because it looks like [this](http://pixel.nymag.com/imgs/daily/entertainment/upload/2010/06/creepy_crossbreed_babies/20100604_crossbreedbabies-alienresurrection.nocrop.w670.h377.jpg). But I can use a crop for other creatures that are not listed. I just have no idea how to do so without a ton of performance issues.

ChangelingRain commented on code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:
```diff
@@ -87,17 +91,72 @@
+	if(state_open)
+		icon_state = "pod-open"
+	else if(occupant)
+		var/mutable_appearance/occupant_overlay
```
You should be using an image here, since mutable appearances don't support having a different dir from the pod, and people should probably usually face SOUTH in cyro pods.

at least, unless you want to call that a feature?

Cyberboss replied:
pods should always be SOUTH though, no?

ChangelingRain replied:
No, they have other dirs so they can connect to pipes from other directions.

RemieRichards replied:
What do you mean they don't support different dirs?
You just do ```occupant_overlay.dir = SOMEDIR```

ChangelingRain replied:
I had the impression from #26518 that mutable appearances used as overlays, like these are, can't have a different dir from the pod itself.

Cyberboss replied:
In that case, go ahead and use images

nicbn replied:
Ok I'll do it.

WJohn commented:
Well, these are better, though they could still use some improvements. I think I'll perhaps try and improve these myself at a later date, but these are fine now 👍

Cyberboss reviewed (changes requested):
See above

nicbn commented:
Should I create new is helpers for alien types or use istype?

Cyberboss commented:
@nicbn

nicbn committed:
Moar code commit, why not?
